python/ql/lib/semmle/python/security/dataflow/TarSlipCustomizations.qll

```diff
@@ -58,20 +92,36 @@ module TarSlip {
   /**
    * A sink capturing method calls to `extractall`.
    *
-   * For a call to `file.extractall` without arguments, `file` is considered a sink.
+   * For a call to `file.extractall` without `members` argument, `file` is considered a sink.
    */
-  class ExtractAllSink extends Sink {
-    ExtractAllSink() {
-      exists(DataFlow::CallCfgNode call |
-        call =
-          API::moduleImport("tarfile")
-              .getMember("open")
-              .getReturn()
-              .getMember("extractall")
-              .getACall() and
-        not exists(call.getArg(_)) and
-        not exists(call.getArgByName(_)) and
-        this = call.(DataFlow::MethodCallNode).getObject()
+  class ExtractAllwoMembersSink extends Sink {
+    ExtractAllwoMembersSink() {
+      exists(AllTarfileOpens atfo, MethodCallNode call |
+        call = atfo.getReturn().getMember("extractall").getACall() and
+        not exists(Node arg | arg = call.getArgByName("members")) and
+        this = call.getObject()
+      )
+    }
+  }
+
+    /**
+   * A sink capturing method calls to `extractall`.
+   *
+   * For a call to `file.extractall` with `members` argument, `file` is considered a sink.
```

Comment:
This does not seem to be what the code does.

Reply:
The comment is updated to: 
```
    /**
   * A sink capturing method calls to `extractall` with `members` arguments.
   *
   * For a call to `file.extractall` with `members` argument, `file` is considered a sink if not
   * a the `members` argument contains a NameConstant as None, a List or call to the method `getmembers`.
   * Otherwise, the argument of `members` is considered a sink.  
   */
```